src/govuk/components/checkboxes/template.njk

```diff
@@ -21,6 +21,13 @@
   {% endif %}
 {% endfor %}
 
+{% set hasBehaviour = false %}
+{% for item in params.items %}
+  {% if item.behaviour %}
+    {% set hasBehaviour = true %}
+  {% endif %}
+{% endfor %}
+
```

Comment:
I am wondering if we could just simplify things by always including the `module` attribute – but if we were going to do that we should do it consistently, so this seems like something we should revisit in the future instead.

Reply:
Good shout to combine them. I'll do that.

I'm guessing that not initialising the javascript when it's not needed saves a tiny bit of performance – but it might be fairly trivial?

Reply:
Done in c401d6cc0770657784f5a21203b2d04991080943

Reply:
> I'm guessing that not initialising the javascript when it's not needed saves a tiny bit of performance – but it might be fairly trivial?

Prompted by @adamsilver's comment I just tested having multiple modules on the page and discovered a bit of an edge case which I think means we need to always include the module attribute.

If you have multiple groups of checkboxes with the same name on the page, but only some of those groups have the exclusive behaviour or a conditional reveal, checking one of the checkboxes in a groups without the module attribute on _does not_ uncheck the exclusive checkbox with the same name, because there are no event listeners set up on the group of checkboxes you checked.

For example, try checking "None of the above" in this example, then checking one of the colours in the other groups.

```jinja
{% call govukFieldset({
    legend: {
    text: "Which colours do you like?",
    classes: "govuk-fieldset__legend--l"
    }
})%}

  <h2 class="govuk-heading-m">Primary colours</h2>

  {{ govukCheckboxes({
    idPrefix: "colour-primary",
    name: "colour",
    items: [
      {
        value: "red",
        text: "Red"
      },
      {
        value: "yellow",
        text: "Yellow"
      },
      {
        value: "blue",
        text: "Blue"
      }
    ]
  }) }}

  <h2 class="govuk-heading-m">Secondary colours</h2>

  {{ govukCheckboxes({
    idPrefix: "colour-secondary",
    name: "colour",
    items: [
      {
        value: "green",
        text: "Green"
      },
      {
        value: "purple",
        text: "Purple"
      },
      {
        value: "orange",
        text: "Orange"
      }
    ]
  }) }}

  <h2 class="govuk-heading-m">Other colours</h2>

  {{ govukCheckboxes({
    idPrefix: "colour-other",
    name: "colour",
    items: [
      {
        value: "imaginary",
        text: "An imaginary colour"
      },
      {
        divider: "or"
      },
      {
        value: "none",
        text: "None of the above",
        behaviour: "exclusive"
      }
    ]
  }) }}

{% endcall %}
```

Reply:
@36degrees nice spot. Shall we remove the logic and always include the `data-module="govuk-radio"` attribute?

Reply:
@36degrees I've fixed this issue by always including the `data-module` attribute here: 0629a0a571baa129bae07ed9b43baa21fe3d97f5